client/src/includes/tabs.js

```diff
@@ -0,0 +1,339 @@
+    this.bindEvents();
+
+    // Set active tab from url or make first tab active
+    if (this.tabButtons) {
+      // Set each button's aria-controls attribute and select tab if aria-selected has already been set on the element
+      this.tabButtons.forEach((button) => {
+        button.setAttribute(
+          'aria-controls',
+          button.getAttribute('href').replace('#', ''),
+        );
+      });
+
+      // Check for active items set by the template
+      const tabActive = [...this.tabButtons].find(
+        (button) => button.getAttribute('aria-selected') === 'true',
+      );
+
+      if (window.location.hash && !this.disableURL) {
+        this.selectTabByURLHash();
+      } else if (tabActive) {
+        // If a tab isn't hidden for some reason hide it
+        this.tabPanels.forEach((tab) => {
+          // eslint-disable-next-line no-param-reassign
+          tab.hidden = true;
+        });
+        // Show aria-selected tab
+        this.selectTab(tabActive);
+      } else {
+        this.selectFirstTab();
+      }
+    }
+  }
+
+  /**
+   * @param {string}newTabId
+   */
+  unSelectActiveTab(newTabId) {
+    // IF new tab ID is the current then don't transition out
+    if (newTabId === this.state.activeTabID || !this.state.activeTabID) {
+      return;
+    }
+
+    // Tab Content to deactivate
+    const tabContent = this.tabContainer.querySelector(
+      `#${this.state.activeTabID}`,
+    );
+
+    if (!tabContent) {
+      return;
+    }
+
+    if (this.animate) {
+      this.animateOut(tabContent);
+    } else {
+      tabContent.hidden = true;
+    }
+
+    const tab = this.tabContainer.querySelector(
+      `a[href='#${this.state.activeTabID}']`,
+    );
+
+    tab.setAttribute('aria-selected', 'false');
+    tab.setAttribute('tabindex', '-1');
+  }
+
+  /**
+   * @param {HTMLElement}tab
+   */
+  selectTab(tab) {
+    if (tab) {
+      const tabContentId = tab.getAttribute('aria-controls');
+
+      // Unselect currently active tab
+      if (tabContentId) {
+        this.unSelectActiveTab(tabContentId);
+      }
+
+      this.state.activeTabID = tabContentId;
+
+      tab.setAttribute('aria-selected', true);
+      tab.removeAttribute('tabindex');
+      const tabContent = this.tabContainer.querySelector(`#${tabContentId}`);
+
+      if (!tabContent) {
+        return;
+      }
+
+      if (this.animate) {
+        this.animateIn(tabContent);
+      } else {
+        tabContent.hidden = false;
+      }
+
+      if (this.state.initialPageLoad) {
+        // On first load set the scroll to top to avoid scrolling to active section and header covering up tabs
+        setTimeout(() => {
+          window.scrollTo(0, 0);
+        }, this.state.transition * 2);
+      }
+
+      // Dispatch tab selected event for the rest of the admin to hook into if needed
+      // Trigger tab specific switch event
+      this.tabList.dispatchEvent(
+        new CustomEvent('switch', { detail: { tab: tab.dataset.tab } }),
+      );
```

Comment:
I see this being fired here but you’ve removed the code that relies on this event, so I’m confused. If we don’t have code that uses this, we should remove this event.

Reply:
I've added the code that relies on this event back for now so might be good to keep it 🤔 now?